test/unit_tests/models/test_modules.py

```diff
@@ -305,3 +329,413 @@ def test_safelog_extra_repr(eps, expected_repr):
+
+    # Define the expected default band_filters
+    expected_band_filters = [(low, low + 4) for low in range(4, 36 + 1, 4)]
+
+    # Assertions to verify band_filters and number of bands
+    assert layer.band_filters == expected_band_filters, "Default band_filters not set correctly when band_filters=None."
+    assert layer.n_bands == 9, "Number of bands should be 9 when band_filters=None."
+
+    # Forward pass to ensure output shape is correct
+    output = layer(sample_input)
+    assert output.shape == (sample_input.shape[0], layer.n_bands, n_chans,
+                            sample_input.shape[
+                                2]), "Output shape is incorrect when band_filters=None."
+
+
+def test_band_filters_with_incorrect_tuple_length():
+    """Test that providing band_filters with tuples not of length 2 raises a ValueError."""
+    n_chans = 8
+    sfreq = 100
+    invalid_band_filters = [(4, 8), (12,)]  # Second tuple has only one element
+
+    with pytest.raises(ValueError,
+                       match="The band_filters items should be splitable in 2 values"):
+        FilterBankLayer(n_chans=n_chans, sfreq=sfreq,
+                        band_filters=invalid_band_filters)
+
+
+def test_iir_params_output_sos_warning(sample_input):
+    """Test that providing iir_params with output='sos' raises a warning and modifies the output."""
+    n_chans = 8
+    sfreq = 100
+    iir_params = {"output": "sos"}
+
+    with pytest.warns(UserWarning,
+                      match="It is not possible to use second-order section"):
+        layer = FilterBankLayer(
+            n_chans=n_chans,
+            sfreq=sfreq,
+            band_filters=None,
+            method="iir",
+            iir_params=iir_params
+        )
+
+    assert layer.filts is not None, "Filters should be initialized."
+    assert layer.filts["band_0"][
+               "a"].dtype == torch.float64, "Filter coefficients should be float64."
+
+
+def test_forward_pass_iir(sample_input):
+    """Test the forward pass of the FilterBankLayer with IIR filtering."""
+    n_chans = 8
+    sfreq = 100
+    iir_params = {"output": "ba"}
+    layer = FilterBankLayer(
+        n_chans=n_chans,
+        sfreq=sfreq,
+        band_filters=None,
+        method="iir",
+        iir_params=iir_params
+    )
+
+    output = layer(sample_input)
+    assert output.shape == (
+    sample_input.shape[0], layer.n_bands, n_chans, sample_input.shape[2]), \
+        "Output shape is incorrect for IIR filtering."
+
+
+def test_forward_pass_fir(sample_input):
+    """Test the forward pass of the FilterBankLayer with FIR filtering."""
+    n_chans = 8
+    sfreq = 100
+    layer = FilterBankLayer(
+        n_chans=n_chans,
+        sfreq=sfreq,
+        band_filters=None,
+        method="fir"
+    )
+
+    output = layer(sample_input)
+    assert output.shape == (
+    sample_input.shape[0], layer.n_bands, n_chans, sample_input.shape[2]), \
+        "Output shape is incorrect for FIR filtering."
```

Comment:
Usually, it is a good convention to start with the simple tests at the top of the file so I would move these tests to the top.

Also, you can merge them:
```suggestion
@pytest.mark.parametrize('method', ['iir', 'fir'])
def test_forward_pass_filter_bank(method, sample_input):
    """Test the forward pass of the FilterBankLayer with IIR filtering."""
    n_chans = 8
    sfreq = 100
    layer = FilterBankLayer(
        n_chans=n_chans,
        sfreq=sfreq,
        band_filters=None,
        method=method,
    )

    output = layer(sample_input)
    assert output.shape == (
        sample_input.shape[0], layer.n_bands, n_chans, sample_input.shape[2]
    ), f"Output shape is incorrect for {method} filtering."
```